### PR TITLE
Multi speed fan power part2 - fix fan power for continuous speed control

### PR DESCRIPTION
**Pull request by EnergyArchmage**
Pull request overview
---------------------
Follow up to PR #6074 , fix fan power calculation problem. 

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by mjwitte** [comment]
@EnergyArchmage Thanks.  I'll compare this branch back to pre-6074.

**Comment by mjwitte** [comment]
@EnergyArchmage This fixes the non-unitarysystem files that have Fan:SystemModel with continuous speed control  That's 7 out of 8 that are showing diffs here - they now match exactly with results prior to any of these multi-speed fan changes.  UnitarySystem_MultiSpeedCoils_SingleMode still has diffs.  Will keep looking at that.

**Comment by mjwitte** [comment]
@EnergyArchmage For UnitarySystem_MultiSpeedCoils_SingleMode debugging revealed mass flow problems which partly stem from the Fan not sizing correctly. In v8.6 with Fan:OnOff, the fan was hard-sized to 1.7 which matches the user-specified unitary system max heating/cooling flow rates.  But the new fan is autosized and it's sizing to 1.37 which matches the unitary system design flow rate (which I'm not sure what that even means:
```
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, Design Size Supply Air Flow Rate [m3/s], 1.37381
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, Design Size Heating Supply Air Flow Rate [m3/s], 1.70000
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, User-Specified Heating Supply Air Flow Rate [m3/s], 1.70000
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, Design Size Cooling Supply Air Flow Rate [m3/s], 1.70000
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, User-Specified Cooling Supply Air Flow Rate [m3/s], 1.70000
 Component Sizing Information, AirLoopHVAC:UnitarySystem, DXAC HEAT PUMP 1, User-Specified No Load Supply Air Flow Rate [m3/s], 0.00000
 Component Sizing Information, Fan:SystemModel, SUPPLY FAN 1, Design Size Design Maximum Air Flow Rate [m3/s], 1.37381
```

So, not sure if the sizing mismatch is a problem with Fan:SystemModel or with AirloopHVAC:UnitarySystem.  For now I'm going to change the example file to hard-size the fan.

**Comment by mjwitte** [comment]
But this also led me to change the arguments passed to the fan to be massFlowRate1/2 instead of flowRatio1/2 because it's not clear that the flow ratios being passed before were always on the same basis if the system, for example, had different heating and cooling max flow rates.

File: src/EnergyPlus/HVACFan.cc [hunk 1]
```diff
@@ -630,6 +635,13 @@ namespace HVACFan {
 		localFlowRatio.resize( 2, 0.0 );
 		localRunTimeFrac.resize( 2, 1.0 );
 
+		Real64 massFlowFromFlowRatios = ( m_maxAirMassFlowRate * ( flowRatio1 * runTimeFrac1 + flowRatio2 * runTimeFrac2 ) );
+		if ( abs ( m_inletAirMassFlowRate - massFlowFromFlowRatios ) > DataHVACGlobals::SmallMassFlow ) {
+			if( !localMismatchErrorShown ) ShowWarningError( "Flow rate mismatch in calcSimpleSystemFan (only first instance is reported)" );
```
Comment (mjwitte) [inline src/EnergyPlus/HVACFan.cc]: This is just temporary to see if any of the test files are sending arguments that don't match the inlet node mass flow rate

File: CMakeLists.txt [hunk 2]
```diff
@@ -84,7 +84,7 @@ if( BUILD_TESTING )
       set( ARCH_FLAG "" )
     endif()
 
-    set( REGRESSION_BASELINE_SHA "" CACHE STRING "SHA of baseline comparison" )
+    set( REGRESSION_BASELINE_SHA "ba9e01f" CACHE STRING "SHA of baseline comparison" )
```
Comment (mjwitte) [inline CMakeLists.txt]: Comparing back to just before the multi-speed fan changes went in

**Comment by mjwitte** [comment]
This should be fixed now.  Also added more unit tests.  Still (attempting to) compare back to pre-fan SHA.  If first CI comes back as expected, then will move up to current develop and push again.

**Comment by EnergyArchmage** [comment]
I reviewed these changes, built locally, ran unit tests locally, reviewed unit tests.  all looks good.

**Comment by mjwitte** [comment]
@EnergyArchmage Checking a few more things on this end.

**Comment by Myoldmopar** [comment]
There is a build warning from this unused variable: https://github.com/NREL/EnergyPlus/blob/b08db0aa95919829e76849ddfcfc1291e30c6b36/src/EnergyPlus/HVACFan.cc#L629

**Comment by mjwitte** [comment]
Good news is that err diffs are getting rid of some errors.
Audit diff is due to hard sized fan in UnitarySystem_MultiSpeedCoils_SingleMode (making it like it was in v8.6)

eio diffs are all minor warmup diffs, except in RefBldgMediumOfficeNew2004_Chicago, not sure how this could impact that:
```
 ! &lt;VAV DX Cooling Coil Standard Rating Information>, DX Coil Type, DX Coil Name, Fan Type, Fan Name, Standard Net Cooling Capacity {W}, Standard Net Cooling Capacity {Btu/h}, IEER {Btu/W-h}, COP 100% Capacity {W/W}, COP 75% Capacity {W/W}, COP 50% Capacity {W/W}, COP 25% Capacity {W/W}, EER 100% Capacity {Btu/W-h}, EER 75% Capacity {Btu/W-h}, EER 50% Capacity {Btu/W-h}, EER 25% Capacity {Btu/W-h}, Supply Air Flow 100% {kg/s}, Supply Air Flow 75% {kg/s},Supply Air Flow 50% {kg/s},Supply Air Flow 25% {kg/s}
- VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,VAV_1_COOLC DXCOIL,Fan:VariableVolume,VAV_1_FAN,140067.28,477929.40,13.96,3.23,3.89,4.48,4.50,11.01,13.26,15.29,15.35,7.1930,5.7294,3.8199,1.7070,
+ VAV DX Cooling Coil Standard Rating Information, Coil:Cooling:DX:TwoSpeed,VAV_1_COOLC DXCOIL,Fan:VariableVolume,VAV_1_FAN,139250.83,475143.55,12.78,2.83,3.49,4.19,4.29,9.64,11.92,14.30,14.63,7.1930,5.7241,3.8122,1.7035,

```

Need to review eso/mtr diffs yet.

**Comment by mjwitte** [comment]
Compared output for this branch with before fan changes https://github.com/NREL/EnergyPlus/commit/ba9e01ff2906492c9728daa7133931ba9bd12735
UnitarySystem_VSHeatPumpWaterToAirEquationFit - tiny diffs
UnitarySystem_MultiSpeedCoils_SingleMode - diffs as expected (partly due to hard-sized fan change, partly due to better fan power calc)
UnitarySystem_HeatPumpAuto - tiny diffs
more later

**Comment by mjwitte** [comment]
UnitarySystem_FurnaceWithDXSystemRHcontrol - tinydiffs
UnitarySystem_5ZoneWaterLoopHeatPump - tiny diffs
StripMallZoneEvapCoolerAutosized - no diffs
RefBldgMediumOfficeNew2004_Chicago - no diffs
PackagedTerminalHeatPumpVSAS - no diffs
OutdoorAirUnit - no diffs
5ZoneSupRetPlenVSATU - no diffs
5ZoneNightVent3 - no diffs

@Myoldmopar You may press the green button whenever you see fit.

**Pull request merged**